libraries/plugins/market_history/market_history_plugin.cpp

```diff
@@ -104,39 +105,54 @@ struct operation_process_fill_order
 
       db.create<order_history_object>( [&]( order_history_object& ho ) {
          ho.key = hkey;
-         ho.time = time;
+         ho.time = _now;
          ho.op = o;
       });
 
-      /*
-      hkey.sequence += 200;
+      // To remove old filled order data
+      const auto max_records = _plugin.max_order_his_records_per_market();
+      hkey.sequence += max_records;
       itr = history_idx.lower_bound( hkey );
-      while( itr != history_idx.end() )
+      if( itr != history_idx.end() && itr->key.base == hkey.base && itr->key.quote == hkey.quote )
       {
-         if( itr->key.base == hkey.base && itr->key.quote == hkey.quote )
+         const auto max_seconds = _plugin.max_order_his_seconds_per_market();
+         fc::time_point_sec min_time;
+         if( min_time + max_seconds < _now )
+            min_time = _now - max_seconds;
+         auto time_itr = his_time_idx.lower_bound( std::make_tuple( hkey.base, hkey.quote, min_time ) );
+         if( time_itr != his_time_idx.end() && time_itr->key.base == hkey.base && time_itr->key.quote == hkey.quote )
          {
-            db.remove( *itr );
-            itr = history_idx.lower_bound( hkey );
+            if( itr->key.sequence >= time_itr->key.sequence )
+            {
+               while( itr != history_idx.end() && itr->key.base == hkey.base && itr->key.quote == hkey.quote )
+               {
+                  auto old_itr = itr;
+                  ++itr;
+                  db.remove( *old_itr );
+               }
+            }
+            else
+            {
+               while( time_itr != his_time_idx.end() && time_itr->key.base == hkey.base && time_itr->key.quote == hkey.quote )
+               {
+                  auto old_itr = time_itr;
+                  ++time_itr;
+                  db.remove( *old_itr );
+               }
+            }
          }
-         else break;
       }
-      */
 
-      /* Note: below is not true, because global settlement creates only one fill_order_op.
-       * for every matched order there are two fill order operations created, one for
-       * each side.  We can filter the duplicates by only considering the fill operations where
-       * the base > quote
-       */
-      /*
-      if( o.pays.asset_id > o.receives.asset_id )
-      {
-         //ilog( "     skipping because base > quote" );
-         return;
-      }
-      */
+      // To update buckets data, only update for maker orders
       if( !o.is_maker )
          return;
 
+      const auto max_history = _plugin.max_history();
+      if( max_history == 0 ) return;
```

Comment:
do we need this check ? i started to test the options with a setup as:

`programs/witness_node/witness_node --data-dir data/my-blockprod --rpc-endpoint "127.0.0.1:8090" --max-ops-per-account 1000 --partial-operations true --max-order-his-seconds-per-market 259200 --max-order-his-records-per-market 0 --history-per-size 0`

to later realize that i had to change `history-per-size` to at least `1` to make it work.

Reply:
There are two sets of data being stored in the market history plugin: the detailed order matching history, and the statistics. `history-per-size == 0` means to not store statistics, but it should not prevent order matching history from being stored. The original code skips statistics data storing if `history-per-size` is set to `0`, but it also skips order matching history storing, which I think is a bug. Now the new options controls how order matching history being stored.

Reply:
ok, got it. thanks for explaining it.